lib/jekyll/filters/url_filters.rb

```diff
@@ -23,9 +23,9 @@ def absolute_url(input)
       # Returns a URL relative to the domain root as a String.
       def relative_url(input)
         return if input.nil?
-        return ensure_leading_slash(input.to_s) if sanitized_baseurl.nil?
+        parts = [sanitized_baseurl, input]
         Addressable::URI.parse(
-          ensure_leading_slash(sanitized_baseurl) + ensure_leading_slash(input.to_s)
+          parts.compact.map { |part| ensure_leading_slash(part.to_s) }.join
```

Comment:
can we have at least one *mutating* method here instead of two non-mutating methods?
i.e. either
```
parts.compact!.map { }
```
or
```
parts.compact.map! { }
```